File: mobile/android/app/src/main/java/dev/alkhalaf/alyamamah/models/Day.kt
package dev.alkhalaf.alyamamah.models

enum class Day(val abbreviation: String) {
    SUN("sun"),
    MON("mon"),
    TUE("tue"),
    WED("wed"),
    THU("thu"),
    FRI("fri"),
    SAT("sat");

    companion object {
        fun fromAbbreviation(abbreviation: String): Day? {
            return values().firstOrNull { it.abbreviation == abbreviation }
        }
    }
}

File: mobile/ios/CoursesWidget/Models/Day.swift
//
//  DayEnum.swift
//  CoursesWidgetExtension
//
//  Created by Yazeed AlKhalaf on 11/03/2023.
//

import Foundation

enum Day: String, Decodable, CodingKey {
    case sun = "sun"
    case mon = "mon"
    case tue = "tue"
    case wed = "wed"
    case thu = "thu"
    case fri = "fri"
    case sat = "sat"
    
    static let values: [Day] = [.sun, .mon, .tue, .wed, .thu]
    static let allValues: [Day] = [.sun, .mon, .tue, .wed, .thu, .fri, .sat]
}
